python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -343,20 +351,28 @@ def match(self, source_array, n_check, n_match, n_agree,
             # store the the rotated versions of our test points for later
             # use.
             tmp_rot_vect_list = []
-            for test_vect in test_vect_list:
-                tmp_rot_vect_list.append(np.dot(shift_rot_matrix, test_vect))
+            for test_vect in test_vectors:
+                tmp_rot_vect_list.append(
+                    np.dot(shift_rot_matrix, test_vect))
+            # Since our test point are in the source frame, we can test if
+            # their lengths are mostly preserved under the transform.
+            if not self._test_pattern_lengths(np.array(tmp_rot_vect_list),
+                                              max_dist_rad):
+                continue
+
             tmp_rot_vect_list.append(pattern_idx)
             rot_vect_list.append(tmp_rot_vect_list)
 
             # Test if we have enough rotations, which agree, or if we
             # are in optimistic mode.
-            if self._test_rotation_agreement(rot_vect_list, max_dist_rad) < \
-               n_agree - 1:
+            if self._test_rotation_agreement(rot_vect_list,
+                                             max_dist_rad) < n_agree - 1:
```

Comment:
One line?

Reply:
Ditto here, though I did change where the line break happens on the latest commit.